src/expandableCalendar/WeekCalendar/index.js

```diff
@@ -100,79 +102,45 @@ class WeekCalendar extends Component {
   });
 
   onDayPress = value => {
-    _.invoke(this.props.context, 'setDate', value.dateString, UPDATE_SOURCES.DAY_PRESS);
+    this.presenter.onDayPressed(this.props.context, value);
   };
 
   onScroll = ({
     nativeEvent: {
       contentOffset: {x}
     }
   }) => {
-    if (!this.firstAndroidRTLScrollIgnored) {
-      this.firstAndroidRTLScrollIgnored = true;
-      return;
-    }
-
-    // Fix reversed offset on Android+RTL
-    if (applyAndroidRtlFix) {
-      const numOfPages = this.state.items.length - 1;
-      const overallWidth = numOfPages * this.containerWidth;
-      x = overallWidth - x;
-    }
-
-    const newPage = Math.round(x / this.containerWidth);
+    const {onScroll} = this.presenter;
+    const {context} = this.props;
+    const {items} = this.state;
+    const {containerWidth, page} = this;
 
-    if (this.page !== newPage) {
-      const {items} = this.state;
+    const updateState = (newData, newPage) => {
       this.page = newPage;
+      this.setState({items: [...newData]});
+    };
 
-      _.invoke(this.props.context, 'setDate', items[this.page], UPDATE_SOURCES.WEEK_SCROLL);
-
-      if (this.page === items.length - 1) {
-        for (let i = 0; i <= NUMBER_OF_PAGES; i++) {
-          items[i] = items[i + NUMBER_OF_PAGES];
-        }
-        this.setState({items: [...items]});
-      } else if (this.page === 0) {
-        for (let i = items.length - 1; i >= NUMBER_OF_PAGES; i--) {
-          items[i] = items[i - NUMBER_OF_PAGES];
-        }
-        this.setState({items: [...items]});
-      }
-    }
+    onScroll({context, updateState, x, page, items, width: containerWidth});
   };
 
   onMomentumScrollEnd = () => {
     const {items} = this.state;
-    const isFirstPage = this.page === 0;
-    const isLastPage = this.page === items.length - 1;
-
-    if (isFirstPage || isLastPage) {
-      this.list.current.scrollToIndex({animated: false, index: NUMBER_OF_PAGES});
-      this.page = NUMBER_OF_PAGES;
-      const newWeekArray = this.getDatesArray();
-
-      if (isLastPage) {
-        for (let i = NUMBER_OF_PAGES + 1; i < items.length; i++) {
-          items[i] = newWeekArray[i];
-        }
-      } else {
-        for (let i = 0; i < NUMBER_OF_PAGES; i++) {
-          items[i] = newWeekArray[i];
-        }
-      }
+    const {onMomentumScrollEnd} = this.presenter;
 
+    const updatedItems = items => {
       setTimeout(() => {
         this.setState({items: [...items]});
       }, 100);
-    }
+    };
+
+    onMomentumScrollEnd({items, props: this.props, page: this.page, updatedItems});
   };
 
   renderItem = ({item}) => {
     const {style, onDayPress, markedDates, ...others} = extractComponentProps(Week, this.props);
-
-    const {context, firstDay} = this.props;
-    const isCurrentWeek = sameWeek(item, context.date, firstDay);
+    const {isCurrentWeek} = this.presenter;
```

Comment:
Where is `isCurrentWeek` on the presenter? I don't see the method...

Reply:
merge issue, fixed